op-acceptance-tests/tests/interop/loadtest/interop_load_test.go

```diff
@@ -67,33 +64,48 @@ func TestLoad(gt *testing.T) {
 	}()
 }
 
+func fundEOAs(num uint64, funder *dsl.Funder) []*dsl.EOA {
+	eoas := make([]*dsl.EOA, 0, num)
+	for range num {
+		eoas = append(eoas, funder.NewFundedEOA(eth.OneEther))
+	}
+	return eoas
+}
+
 func SpamInteropTxs(t devtest.T, numInitTxs uint64, source *L2, dest *L2, supervisor *dsl.Supervisor) {
 	var wg sync.WaitGroup
 	defer wg.Wait()
 	msgsCh := make(chan []types.Message, 100)
 	defer close(msgsCh)
 
+	// Mempool implementations may limit the number of concurrent transactions per account.
+	// We spam transactions from multiple EOAs to mitigate the possibility of mempool
+	// implementations being a limiting factor.
+
 	// Spam executing messages.
 	wg.Add(1)
 	go func() {
 		defer wg.Done()
-		dest.Funder.NewFundedEOA(eth.MillionEther.Mul(100))
 		relayers := []Relayer{
-			NewValidRelayer(dest.Funder, dest.EL, supervisor),
-			NewDelayedRelayer(NewValidRelayer(dest.Funder, dest.EL, supervisor), &wg, time.Minute),
-			NewInvalidRelayer(dest.Funder, dest.EL, makeInvalidChainID),
-			NewInvalidRelayer(dest.Funder, dest.EL, makeInvalidBlockNumber),
-			NewInvalidRelayer(dest.Funder, dest.EL, makeInvalidLogIndex),
-			NewInvalidRelayer(dest.Funder, dest.EL, makeInvalidOrigin),
-			NewInvalidRelayer(dest.Funder, dest.EL, makeInvalidPayloadHash),
-			NewInvalidRelayer(dest.Funder, dest.EL, makeInvalidTimestamp),
+			NewValidRelayer(dest.EL, supervisor),
+			NewDelayedRelayer(NewValidRelayer(dest.EL, supervisor), &wg, time.Minute),
+			NewInvalidRelayer(dest.EL, makeInvalidChainID),
+			NewInvalidRelayer(dest.EL, makeInvalidBlockNumber),
+			NewInvalidRelayer(dest.EL, makeInvalidLogIndex),
+			NewInvalidRelayer(dest.EL, makeInvalidOrigin),
+			NewInvalidRelayer(dest.EL, makeInvalidPayloadHash),
+			NewInvalidRelayer(dest.EL, makeInvalidTimestamp),
```

Comment:
might be nice to generalize these into a single `NewRelayer(opts ...relayerOpt)` where you pass in different options depending on what you want to do.

So for invalid relayers you'd do

    NewRelayer(dest.EL, WithInvalidChainID())
    
@protolambda's PR here shows how you might implement these patterns: https://github.com/ethereum-optimism/optimism/pull/15720

Not a priority but something to keep in mind.

Reply:
Thanks. I'll make a note of this and implement the refactor when things have stabilized a bit more.